lib/district_repository.rb

```diff
@@ -14,47 +12,59 @@ def find_by_name(name)
     pair = districts.detect do |_key, value|
       name.strip.upcase == value.name.upcase
     end
+
     pair && pair[1]
   end
 
-  def find_all_matching(name_fragment)
+  def find_all_districts_matching_name_fragment(name_fragment)
     matching = districts.select do |_key, value|
       value.name.upcase.include?(name_fragment.upcase)
     end
+
     matching.values
   end
 
-  def parser(contents)
+  def kindergarten_participation_access_hash
+    path = "data/Kindergartners in full-day program.csv"
+    { :kindergarten_participation => path }
+  end
+
+  def hs_graduation_rate_access_hash
+    path = "data/High school graduation rates.csv"
+    { :high_school_graduation_rates => path }
+  end
+
+  def create_enrollment_repository
+    # create enrollment repo to fetch data from
+    file_paths = [kindergarten_participation_access_hash,
+                  hs_graduation_rate_access_hash]
+
+    @enrollment_repository = EnrollmentRepository.new
+    @enrollment_repository.load_enrollment_data(file_paths)
+  end
+
+  def load_district_data(csv_filepath)
```

Comment:
This method was disassembled slightly but comments were added in place of vague method names, since what we're doing here is pushing districts into what turns out to be a collection.